lib/improver/tests/nowcasting/accumulation/test_Accumulation.py

```diff
@@ -168,97 +177,224 @@ def test_times(self):
+        cubes, so the integral is 5."""
+        expected_cube_subset = self.cubes[:6]
+        upper_bound_fp, = self.cubes[5].coord("forecast_period").points
+        integral = 5
+        plugin = Accumulation(
+            accumulation_period=5*60,
+            forecast_periods=np.array([5])*60)
+        result = plugin._get_cube_subsets(
+            self.cubes, upper_bound_fp, integral)
+        self.assertEqual(expected_cube_subset, result)
+
+    @ManageWarnings(record=True)
+    def test_partial_period(self, warning_list=None):
+        """Test that if the forecast period of the upper bound cube is
+        not within the list of requested forecast periods, then the
+        subset of cubes returned is equal to None."""
+        expected_cube_subset = None
+        expected_warning_msg = (
             "The provided cubes result in a partial period given the specified"
             " accumulation_period, i.e. the number of cubes is insufficient to"
             " give a set of complete periods. Only complete periods will be"
             " returned.")
-
-        expected = [self.cubes[0:4], self.cubes[3:7], self.cubes[6:10]]
-
-        plugin = Accumulation(accumulation_period=180)
-        result = plugin._get_period_sets(time_interval, self.cubes)
-
-        for index, sublist in enumerate(result):
-            self.assertSequenceEqual(sublist, expected[index])
+        upper_bound_fp = self.cubes[5].coord("forecast_period").points
+        integral = 0
+        plugin = Accumulation(
+            accumulation_period=5*60,
+            forecast_periods=np.array([5])*60)
+        result = plugin._get_cube_subsets(
+            self.cubes, upper_bound_fp, integral)
+        self.assertEqual(expected_cube_subset, result)
         self.assertTrue(any(item.category == UserWarning
                             for item in warning_list))
-        self.assertTrue(any(warning_msg in str(item)
+        self.assertTrue(any(expected_warning_msg in str(item)
                             for item in warning_list))
 
-    def test_raises_exception_for_impossible_aggregation(self):
-        """Test function raises an exception when attempting to create an
-        accumulation_period that cannot be created from the input cubes."""
 
-        time_interval = 61
-        plugin = Accumulation(accumulation_period=120)
-        msg = "The specified accumulation period "
+class Test__calculate_accumulation(rate_cube_set_up):
 
-        with self.assertRaisesRegex(ValueError, msg):
-            plugin._get_period_sets(time_interval, self.cubes)
+    """Test the _calculate_accumulation method."""
+
+    def test_basic(self):
+        """Check the calculations of the accumulations, where an accumulation
+        is constructed from the cube subset by using 50% of the precipitation
+        rates cubes that bookend the accumulation period and 100% of the
+        precipitation rates cubes that are fully within the accumulation
+        period."""
```

Comment:
???? We should chat about this.

Reply:
I've rewritten this docstring to better describe the function being tested.

Reply:
Thanks. I think I get what you were trying to say before, but I prefer the new docstring.